### DISCUSSION TITLE
ContextMenu.ItemsSource binding acts as OneTime even if specified otherwise

### DISCUSSION THREAD

**Discussion by derekantrican** [AvaloniaUI/Avalonia#18444] [Q&A]
### Describe the bug

I have bound a ContextMenu.ItemsSource property to a list in my viewmodel, but the getter is only called once, even if I specify a different binding mode such as OneWay or TwoWay, the bindind still acts as OneTime

### To Reproduce

MainWindow.axaml:
```xml
<Button>
  <Button.ContextMenu>
    <ContextMenu ItemsSource="{Binding Items, Mode=TwoWay}">
      <ContextMenu.Styles>
        <Style Selector="MenuItem">
          <Setter Property="Header" Value="{Binding .}"/>
        </Style>
      </ContextMenu.Styles>
  </Button.ContextMenu>
</Button>
```

MainViewModel.cs:
```csharp
public ObservableCollection<string> Items
{
  get
  { // <---- Placing breakpoint here
    return new ObservableCollection<string>(new List<string> { "Item1", "Item2", "Item3" });
  }
}
```

### Expected behavior

I would expect the getter for `Items` to be called every time the context menu is opened, but it is not

### Avalonia version

11.1.0

### OS

Windows

**Comment by anovik** [comment]
Found related problem: it seems the properties like IsVisible or IsEnabled also has only one-time binding for ContextMenu/MenuItem which is odd. 

Temporarily I solved that by switching from ContextMenu to MenuFlyout, binding there works correctly. But still, there is a problem with ContextMenu.

**Comment by derekantrican** [comment]
@anovik I'm not seeing correct bindings, even with MenuFlyout - the getter is still only called once at the beginning. Can you share your snippet?

Here's what I tried changing mine to:
```xml
<Button>
	<Button.ContextFlyout>
		<MenuFlyout ItemsSource="{Binding Items, Mode=TwoWay}"><!--I tried multiple "Modes" here-->
			<MenuItem Header="{Binding Header}" Command="{Binding ActionCommand}"/>
		</MenuFlyout>
	</Button.ContextFlyout>
</Button>
```

**Comment by derekantrican** [comment]
Ok, ChatGPT to the rescue. I added `AttachedToVisualTree="ContextMenu_AttachedToVisualTree"` to my ContextMenu and in the code-behind, simply called my "LoadMenuItems" command:
```csharp
private void ContextMenu_AttachedToVisualTree(object sender, VisualTreeAttachmentEventArgs e)
{
	mainViewModel.ContextMenuItems.Clear();

        foreach (.....)
	{
		mainViewModel.ContextMenuItems.Add(.....);
	}
}
```

And that seems to dynamically work for updating the items every time. Definitely a workaround - wish I didn't have to use the code-behind for this

**Comment by maxkatz6** [comment]
"OneWay" doesn't mean that it will be called each time control is displayed. 
"OneWay" means value will be updated each time (view) model notifies about changes.
